pilot/pkg/xds/xdsgen.go

```diff
@@ -124,16 +124,26 @@ func (s *DiscoveryServer) pushXds(con *Connection, push *model.PushContext,
 		return err
 	}
 
-	// Some types handle logs inside Generate, skip them here
-	if !gen.Metadata().LogsDetails {
-		if log.DebugEnabled() {
-			// Add additional information to logs when debug mode enabled
-			log.Infof("%s: PUSH%s for node:%s resources:%d size:%s nonce:%v version:%v",
-				v3.GetShortType(w.TypeUrl), req.PushReason(), con.ConID, len(res), util.ByteCount(configSize), resp.Nonce, resp.VersionInfo)
-		} else {
-			log.Infof("%s: PUSH%s for node:%s resources:%d size:%s",
-				v3.GetShortType(w.TypeUrl), req.PushReason(), con.ConID, len(res), util.ByteCount(configSize))
+	ptype := "PUSH"
+	info := ""
+	if logdata.Incremental {
+		ptype = "PUSH INC"
+	}
+	if len(info) > 0 {
+		info = " " + logdata.AdditionalInfo
+	}
+
+	if log.DebugEnabled() {
+		// Add additional information to logs when debug mode enabled
+		log.Infof("%s: %s%s for node:%s resources:%d size:%s nonce:%v version:%v%s",
+			v3.GetShortType(w.TypeUrl), ptype, req.PushReason(), con.ConID, len(res), util.ByteCount(configSize), resp.Nonce, resp.VersionInfo, info)
+		if logdata.Incremental {
+			log.Debugf("%s: %s%s for node:%s resources:%d size:%s%s",
+				v3.GetShortType(w.TypeUrl), ptype, req.PushReason(), con.ConID, len(res), util.ByteCount(configSize), info)
 		}
+	} else {
+		log.Infof("%s: %s%s for node:%s resources:%d size:%s%s",
+			v3.GetShortType(w.TypeUrl), ptype, req.PushReason(), con.ConID, len(res), util.ByteCount(configSize), info)
```

Comment:
```suggestion
	if log.DebugEnabled() && logdata.Incremental {
		log.Debugf("%s: %s%s for node:%s resources:%d size:%s%s",
			v3.GetShortType(w.TypeUrl), ptype, req.PushReason(), con.ConID, len(res), util.ByteCount(configSize), info)
	} else if log.DebugEnabled() {
		// Add additional information to logs when debug mode enabled
		log.Infof("%s: %s%s for node:%s resources:%d size:%s nonce:%v version:%v%s",
			v3.GetShortType(w.TypeUrl), ptype, req.PushReason(), con.ConID, len(res), util.ByteCount(configSize), resp.Nonce, resp.VersionInfo, info)
	} else {
		log.Infof("%s: %s%s for node:%s resources:%d size:%s%s",
			v3.GetShortType(w.TypeUrl), ptype, req.PushReason(), con.ConID, len(res), util.ByteCount(configSize), info
```

I think right now we would be double logging EDS pushes